Frameworks/CoreGraphics/CGFont.mm

```diff
@@ -16,615 +16,134 @@
-}
-
-/**
- @Status Interoperable
-*/
-CFDataRef CGFontCopyTableForTag(CGFontRef font, uint32_t tag) {
-    //  Get the font
-    FT_Face face = (FT_Face)(DWORD)[font _sizingFontHandle];
-
-    _CGFontLock();
-    FT_Error err;
-
-    DWORD numBytes = 0;
-    err = FT_Load_Sfnt_Table(face, tag, 0, NULL, &numBytes);
-
-    char* pData = (char*)IwMalloc(numBytes);
-    err = FT_Load_Sfnt_Table(face, tag, 0, (FT_Byte*)pData, &numBytes);
-
-    id ret = [[NSData alloc] initWithBytes:pData length:numBytes];
-    IwFree(pData);
-
-    _CGFontUnlock();
-
-    return (CFDataRef)ret;
-}
-
-/**
- @Status Caveat
- @Notes Returns family name
-*/
-CFStringRef CGFontCopyFullName(CGFontRef font) {
-    //  Get the font
-    FT_Face face = (FT_Face)(DWORD)[font _sizingFontHandle];
-
-    if ((face == NULL) || (face->family_name == NULL)) {
-        return nullptr;
-    }
-
-    return (CFStringRef)[[NSString stringWithCString:face->family_name] retain];
-}
-
-/**
- @Status Interoperable
-*/
-CGFontRef CGFontCreateWithDataProvider(CGDataProviderRef cgDataProvider) {
-    CFRetain(cgDataProvider);
-    return (CGFontRef)[[_LazyUIFont fontWithData:(NSData*)cgDataProvider] retain];
-}
+static const CFRuntimeClass __CGFontClass = { 0,
+                                              "CGFont",
+                                              __CGFontInit, // init
+                                              NULL, // copy
+                                              __CGFontDeallocate, // deallocate
+                                              __CGFontEqual,
+                                              __CGFontHash,
+                                              NULL, //
+                                              __CGFontCopyDescription };
 
 /**
  @Status Interoperable
 */
 void CGFontRelease(CGFontRef font) {
-    [(id)font release];
+    CFRelease(font);
 }
 
 /**
  @Status Interoperable
 */
 CGFontRef CGFontRetain(CGFontRef font) {
-    [(id)font retain];
-
+    CFRetain(font);
     return font;
 }
 
 /**
- @Status Interoperable
-*/
-int CGFontGetUnitsPerEm(CGFontRef font) {
-    //  Get the font
-    FT_Face face = (FT_Face)[font _sizingFontHandle];
-    return face->units_per_EM;
-}
-
-/**
- @Status Interoperable
+ @Status Stub
 */
-int CGFontGetXHeight(CGFontRef font) {
-    return (int)[font xHeight];
+CGFontRef CGFontCreateWithDataProvider(CGDataProviderRef cgDataProvider) {
+    UNIMPLEMENTED();
+    return StubReturn();
 }
 
 /**
  @Status Interoperable
 */
 CGFontRef CGFontCreateWithFontName(CFStringRef name) {
-    return (CGFontRef)[[_LazyUIFont fontWithName:(NSString*)name size:1.0f] retain];
-}
-
-/**
- @Status Interoperable
-*/
-bool CGFontGetGlyphAdvances(CGFontRef font, const CGGlyph* glyphs, size_t count, int* advances) {
-    DWORD i;
+    size_t memSize = sizeof(struct __CGFont) - sizeof(CFRuntimeBase);
+    CGFontRef ret = static_cast<CGFontRef>(_CFRuntimeCreateInstance(kCFAllocatorDefault, CGFontGetTypeID(), memSize, NULL));
```

Comment:
>ret [](start = 14, length = 3)

use a RAII construct to avoid ugly release logic that is easy to miss when adding code later.